backend/courses/filters.py

```diff
@@ -245,14 +264,24 @@ def get_schema_operation_parameters(self, view):
                 },
             },
             {
-                "name": "requirements",
+                "name": "pre_ngss_requirements",
                 "required": False,
                 "in": "query",
-                "description": "Filter courses by comma-separated requirements, ANDed together. "
-                "Use `/requirements` endpoint to get requirement IDs.",
+                "description": "Deprecated since 2022C. Filter courses by comma-separated pre"
+                "ngss requirements, ANDed together. Use `/requirements` endpoint"
+                "to get requirement IDs.",
                 "schema": {"type": "string"},
                 "example": "SS@SEAS,H@SEAS",
             },
+            {
+                "name": "attributes",
+                "required": False,
+                "in": "query",
+                "description": "Filter courses by comma-separated attributes, ORed together."
```

Comment:
This is something that I was debating, and I would like to get your thoughts on it.

[Attributes](https://docs.google.com/spreadsheets/d/1roS1aypEJx5q3TY6Kj28t70apjUozSxN/edit?usp=sharing&ouid=110074493394863491185&rtpof=true&sd=true) typically identify a subset of courses that fill a requirement for a major/program. In general, there are a few different types
- **Grade Mode** - ex: pass/fail, multi-term course
- **General Education** - the requirements to receive a particular degree (like a B.S. engineering or GSE PHD). ex: SEAS social science, SEAS humanities
- **Major/Program Specifics** - used heavily by the college and Wharton for majors/concentrations
The motivating example for OR is basically wanting to fill a spot in your schedule with some class that fills one of your requirement. For example, an engineering student can identify a class that's either SS or H. On the other hand, AND is useful for identifying double counting courses. In general, double counting is mainly relevant for dual degree or major students (not for minors, since they are not well represented in the attributes). 

Therefore, the decision based around the judgement that the single-degree majority would get more benefit from OR.

Reply:
Hmm yeah this is tough. I think users would expect AND behavior across different attribute types (e.g. if they select pass/fail + SEAS social science, they wouldn't expect it to show all courses that are pass/fail OR social science). But they would probably expect OR behavior within a type, especially if multiple selections are inherently contradictory (e.g. selecting pass/fail and some other grade mode). We should probably leave it up to the frontend to determine how they want to logically connect the query terms, so I just committed a change that affords a more sophisticated query language (consisting of attribute names, ands, ors, and negations) for this filter.